SS_Activation_OpenMC.py

```diff
@@ -0,0 +1,163 @@
+settings.particles = 10000
+settings.source = openmc.Source(space=PointSource, energy=Prob, strength = 1.0, particle = 'neutron')
+settings.run_mode = 'fixed source'
+settings.export_to_xml()
+
+# Define tallies
+neutron_tally = openmc.Tally(name="Neutron tally")
+neutron_tally.scores = ['flux', 'elastic', 'absorption']
+# Implementing filter for neutron tally through W shell
+cell_filter = openmc.CellFilter([Shell])
+neutron_tally.filters = [cell_filter]
+
+# Creating a tally to get the flux energy spectrum.
+# An energy filter is created to assign to the flux tally using the Vitamin J structure.
+energy_filter_flux = openmc.EnergyFilter.from_group_structure("VITAMIN-J-175")
+
+spectrum_tally = openmc.Tally(name="Flux spectrum")
+# Implementing energy and cell filters for flux spectrum tally
+spectrum_tally.filters = [energy_filter_flux, cell_filter]
+spectrum_tally.scores = ['flux']
+
+# Collecting and exporting tallies to .xml
+tallies = openmc.Tallies([neutron_tally, spectrum_tally])
+tallies.export_to_xml()
+
+model = openmc.model.Model(geometry=geometry,settings=settings)
+#Depletion calculation
+W.depletable = True
+W.volume = 4.0/3.0 * np.pi * (R_2**3 - R_1**3) #volume of W wall material
+operator = openmc.deplete.CoupledOperator(model, normalization_mode='source-rate')
+time_steps = [3e8, 86400, 2.6e6]
+source_rates = [1E+18, 0, 0]
+integrator = openmc.deplete.PredictorIntegrator(operator=operator, timesteps=time_steps, source_rates=source_rates, timestep_units='s')
+integrator.integrate()
+
+#Opening statepoint file to read tallies:
+with openmc.StatePoint('statepoint.10.h5') as sp:
+    fl = sp.get_tally(name="Flux spectrum")
+    nt = sp.get_tally(name="Neutron tally")
+    
+# Get the neutron energies from the energy filter
+energy_filter_fl = fl.filters[0]
+energies_fl = energy_filter_fl.bins[:, 0]
+
+# Get the neutron flux values
+flux = fl.get_values(value='mean').ravel()
+
+#Neutron flux/elastic/absorption tallies:
+tal = nt.get_values(value='mean').ravel()
+print(tal)
+
+Flux_Data = np.c_[energies_fl, flux]
+#Creating an excel file that stores flux data for each energy bin (used as input for ALARA)
+#Dividing by volume to obtain proper units of flux (#/cm^2-s)
+Flux_Data = np.c_[energies_fl, flux/W.volume]
+#ALARA flux inputs go from high energy to low energy
+Flux_Data_ALARA = Flux_Data[::-1]
+FD_CSV = pd.DataFrame(Flux_Data_ALARA, columns=['Energy [eV]', 'Flux [n-cm/sp]'])
+FD_CSV.to_csv('Neutron_Flux.csv', index=False)
+
+Tallies_CSV = pd.DataFrame(tal)
+#Creating a csv file that stores total tally value data
+Tallies_CSV.to_csv('Tally_Values.csv', index=False)
+
+# Depletion results file
+results = openmc.deplete.Results(filename='depletion_results.h5')
+
+# Stable W nuclides present at beginning of operation (will not be plotted)
+stable_nuc = ['W180', 'W182', 'W183', 'W184', 'W186']  
+
+#Store list of nuclides from last timestep as a Materials object
+materials_last = results.export_to_materials(-1)
```

Comment:
If you only plot the nuclides from the last time step, you might get relatively stable nuclides and might not see much decay.

Reply:
The nuclides in the last timestep include the nuclides created at all earlier timesteps.

Reply:
However, while I was playing around with the time_steps I found out something interesting. With the current operating time of 3e8 s, around 60 or so nuclides are at the last time step. If a small timestep is added at the beginning (e.g. 1000s) with the same neutron source rate, the number of nuclides at the end is around 230.

Reply:
This seems to suggest that OpenMC isn't tracking the short-lived nuclides that are created near the beginning of operation, unless that short timestep is added in explicitly. Perhaps we should break up the operating time into smaller chunks to account for this